### PR TITLE
update finalzier to CRD doc

### PR DESCRIPTION
**Pull request by stuggi**
Depends-On: https://github.com/openstack-k8s-operators/lib-common/pull/519
Depends-On: https://github.com/openstack-k8s-operators/mariadb-operator/pull/231

### CODE REVIEW

File: tests/kuttl/tests/deploy/10-assert-deploy-ironic.yaml [hunk 1]
```diff
@@ -141,7 +141,7 @@ apiVersion: ironic.openstack.org/v1beta1
 kind: IronicNeutronAgent
 metadata:
   finalizers:
-  - IronicNeutronAgent
+  - openstack.org/ironicneutronagent
```
Comment (abays) [inline tests/kuttl/tests/deploy/10-assert-deploy-ironic.yaml:144]: Was this changed in the Golang code as well?  I'm having trouble seeing it

Comment (stuggi) [reply tests/kuttl/tests/deploy/10-assert-deploy-ironic.yaml:144]: not go code change is required. its the default finalizer it gets from lib-common https://github.com/openstack-k8s-operators/ironic-operator/blob/main/controllers/ironicneutronagent_controller.go#L196

**Review by abays — APPROVED** [review]
/lgtm

**Pull request merged**